### PR TITLE
Custom "from" and authenticated email

### PR DESCRIPTION
**Pull request by skim-milk**
Amazon SES requires sending from an approved address. "requests@<server_name>" is pretty limited, so I added in an env var for a user to supply this value. :')

Let me know if something fails and I'll get it fixed up :)

### CODE REVIEW

**Comment by grahamgilbert** [comment]
Can you add this as an environment variable for the Docker image as well please in https://github.com/grahamgilbert/Crypt-Server/blob/master/docker/settings_import.py ?

**Comment by skim-milk** [comment]
Done! Thanks for pointing that one out.

Isn't the default being set here? https://github.com/grahamgilbert/Crypt-Server/blob/master/server/views.py#L290-L292

**Comment by grahamgilbert** [comment]
Yes, but if you are exposing this as a setting, it is much more readable to set a default setting in settings.py that can then be overridden.

**Comment by skim-milk** [comment]
I threw in an else under settings_import.py, but I feel like I'm misunderstanding this. Feel free to merge

Since this relies on HOST_NAME (which isn't required to be set), a user could just end up with `crypt@` as their EMAIL_SENDER.

This isn't a really an issue because HOST_NAME is only really set for email contents.

Using the system `hostname` isn't really any better, because that won't (99% of the time) won't match up with the resolvable URL in front of the load balancers.

**Comment by skim-milk** [comment]
Added logs and authenticated mail for requesting a secret as well as request approval.
Log config to remove filters came straight from SO 🙏

File: server/views.py [hunk 1]
```diff
@@ -18,10 +19,11 @@
 from django.urls import reverse
 
 # Create your views here.
+logger = logging.getLogger(__name__)
 
 ##clean up old requests
 def cleanup():
-    how_many_days = 7
+    how_many_days = 180
```
Comment (grahamgilbert) [inline server/views.py:26]: Why are we changing this? I don’t think this should be this long. Doesn’t seem related to the PR.

File: server/views.py [hunk 2]
```diff
@@ -288,21 +290,39 @@ def request(request, secret_id):
...
-                                send_mail(
-                                    "Crypt Key Request",
-                                    email_message,
-                                    email_sender,
-                                    [user.email],
-                                    fail_silently=True,
-                                )
-
+                                
+                                logger.error("[!] Sending request email to {} from {}".format(user.email, email_sender))
```
Comment (grahamgilbert) [inline server/views.py:301]: I doubt these should be logged as an error

File: server/views.py [hunk 3]
```diff
@@ -288,21 +290,39 @@ def request(request, secret_id):
...
+                                if settings.EMAIL_USER and settings.EMAIL_PASSWORD:
+                                    
+                                    authing_user = settings.EMAIL_USER
+                                    authing_password = settings.EMAIL_PASSWORD
+                                    logger.error("[!] Authing to mail server as {}".format(authing_user))
```
Comment (grahamgilbert) [inline server/views.py:306]: Same here. This should be info, surely.

**Review by grahamgilbert — CHANGES_REQUESTED** [review]

File: server/views.py (hunk 1 above)
Comment (skim-milk) [reply server/views.py]: Doesn't need to be. I gotta stop working all night :')

File: server/views.py (hunk 2 above)
Comment (skim-milk) [reply server/views.py]: Yeah, I did it to _very much definitely_ see the error. Agreed!

**Comment by skim-milk** [comment]
Completely forgot about this. Looks like I made requested changes - anything else needed for me to review?

**Review by grahamgilbert — APPROVED** [review]

**Pull request merged**